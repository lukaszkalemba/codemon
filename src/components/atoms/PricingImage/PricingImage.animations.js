const imageVariants = {
  initial: { opacity: 0 },
  animate: {
    x: 0,
    opacity: 1,
    transition: {
      duration: 0.75,
    },
  },
};

const imageShadowVariants = {
  initial: { opacity: 0, x: -40, y: -40 },
  animate: {
    opacity: 1,
    x: 0,
    y: 0,
    transition: {
      duration: 1,
    },
  },
};

export default { imageVariants, imageShadowVariants };
