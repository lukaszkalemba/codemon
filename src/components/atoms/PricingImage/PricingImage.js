import React from 'react';
import { graphql, useStaticQuery } from 'gatsby';
import S from './PricingImage.styles';
import animations from './PricingImage.animations';

const IMAGE_QUERY = graphql`
  {
    officeImage: file(relativePath: { eq: "photo_office.png" }) {
      childImageSharp {
        fluid {
          ...GatsbyImageSharpFluid_withWebp_noBase64
        }
      }
    }
  }
`;

const PricingImage = () => {
  const { officeImage } = useStaticQuery(IMAGE_QUERY);

  return (
    <S.ImageWrapper {...animations.image}>
      <S.Image fluid={officeImage.childImageSharp.fluid} />
      <S.ImageShadow {...animations.imageShadow} />
    </S.ImageWrapper>
  );
};

export default PricingImage;
