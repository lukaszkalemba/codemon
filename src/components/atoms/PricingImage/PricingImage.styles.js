import styled from 'styled-components';
import { motion } from 'framer-motion';
import GatsbyImage from 'gatsby-image';

const ImageWrapper = styled(motion.div)`
  display: none;
  position: relative;

  ::before {
    content: '';
  }

  ${({ theme }) => theme.media.laptop`
    display: block;
  `}
`;

const Image = styled(GatsbyImage)`
  margin: 0 40px 40px 0;
  z-index: ${({ theme }) => theme.zindex.level1};
`;

const ImageShadow = styled(motion.div)`
  position: absolute;
  top: 0;
  left: 0;
  width: calc(100% - 40px);
  height: calc(100% - 40px);
  margin: 40px 0 0 40px;
  background-color: ${({ theme }) => theme.color.dark};
`;

export default {
  ImageWrapper,
  Image,
  ImageShadow,
};
