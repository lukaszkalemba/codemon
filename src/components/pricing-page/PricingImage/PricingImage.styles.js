import styled from 'styled-components';
import { motion } from 'framer-motion';
import GatsbyImage from 'gatsby-image';

const ImageWrapper = styled(motion.div)`
  display: none;
  position: relative;

  ::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    width: calc(100% - 40px);
    height: calc(100% - 40px);
    margin: 40px 0 0 40px;
    background-color: ${({ theme }) => theme.color.dark};
  }

  ${({ theme }) => theme.media.laptop`
    display: block;
  `}
`;

const Image = styled(GatsbyImage)`
  margin: 0 40px 40px 0;
`;

export default {
  ImageWrapper,
  Image,
};
