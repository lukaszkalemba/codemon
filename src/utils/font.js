export default {
  size: {
    xxxs: '1rem',
    xxs: '1.3rem',
    xs: '1.6rem',
    s: '2.2rem',
    m: '3rem',
    l: '4rem',
    xl: '5.4rem',
    xxl: '8.5rem',
    xxxl: '10rem',
  },

  weight: {
    regular: '400',
    medium: '500',
    bold: '700',
  },

  family: {
    montserrat: 'Montserrat',
  },
};
