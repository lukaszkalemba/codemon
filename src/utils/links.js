export default [
  {
    id: 0,
    path: '/projekty',
    name: 'projekty',
  },
  {
    id: 1,
    path: '/blog',
    name: 'blog',
  },
  {
    id: 2,
    path: '/wycena',
    name: 'wycena',
  },
];
