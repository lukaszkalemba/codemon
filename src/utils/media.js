export default {
  phone: '576px',
  tablet: '768px',
  desktop: '1150px',
  largeDesktop: '1600px',
};
