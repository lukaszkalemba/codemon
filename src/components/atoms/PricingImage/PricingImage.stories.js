import React from 'react';
import PricingImage from './PricingImage';

export const pricingImage = () => <PricingImage />;

export default {
  component: PricingImage,
  title: 'Atoms|PricingImage',
};
