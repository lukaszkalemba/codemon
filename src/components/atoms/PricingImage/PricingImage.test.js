import React from 'react';
import { render } from 'utils/test-utils';
import * as Gatsby from 'gatsby';
import PricingImage from './PricingImage';

beforeEach(() => {
  const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');

  useStaticQuery.mockImplementation(() => ({
    officeImage: {
      childImageSharp: {
        fluid: {
          aspectRatio: 0,
          src: '',
          srcSet: '',
          sizes: '',
        },
      },
    },
  }));
});

describe('<PricingImage />', () => {
  it('matches snapshot', () => {
    const { container } = render(<PricingImage />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
