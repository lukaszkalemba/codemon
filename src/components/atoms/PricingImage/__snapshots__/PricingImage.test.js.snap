// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PricingImage /> matches snapshot 1`] = `
<div
  class="sc-AxjAm rIIiZ"
  style="opacity: 0;"
>
  <div
    class="sc-AxirZ jXEvzR gatsby-image-wrapper"
    style="position: relative; overflow: hidden;"
  >
    <div
      aria-hidden="true"
      style="width: 100%;"
    />
    <picture>
      <source
        sizes=""
        srcset=""
      />
      <img
        alt=""
        loading="lazy"
        sizes=""
        src=""
        srcset=""
        style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
      />
    </picture>
    <noscript>
      &lt;picture&gt;&lt;source srcset="" /&gt;&lt;img loading="lazy" src="" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
    </noscript>
  </div>
  <div
    class="sc-AxiKw jIxHFw"
    style="opacity: 0; transform: translateX(-40px) translateY(-40px) translateZ(0);"
  />
</div>
`;
